apache/trafficcontrol
# Renaming a host in TC, does not impact hash_id

## What does this PR (Pull Request) do?
- [x] This PR fixes #2156
- On server create, PR sets xmppId = UUID and removes hostname from the equation. 
- On server update, PR prevents xmppId (removing it from update sql query) from ever being changed once set as it is the source of hashId used by traffic router for consistent hashing.

## Which Traffic Control components are affected by this PR?
- Documentation
- Traffic Ops

## What is the best way to verify this PR?
1. Create a server in traffic_portal. Observe the xmpp_id value in db. It should be set to uuid.
2. Edit the hostname of the created server (from step 1)
3. Observe the change of hostname in DB but xmpp_id remains unaffected.

## If this is a bug fix, what versions of Traffic Control are affected?
- master 
- 4.1.0

## The following criteria are ALL met by this PR

## Additional Information
A new package, uuid, was added to traffic_ops_golang vendor folder.
